packages/tenant-process/src/services/tenantService.ts

```diff
@@ -260,6 +273,116 @@ export function tenantServiceBuilder(
       }
     },
 
+    async addCertifiedAttribute(
+      tenantId: TenantId,
+      {
+        tenantAttributeSeed,
+        authData,
+        correlationId,
+      }: {
+        tenantAttributeSeed: ApiCertifiedTenantAttributeSeed;
+        authData: AuthData;
+        correlationId: string;
+      }
+    ): Promise<Tenant> {
+      logger.info(
+        `Add certified attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+      const organizationId = authData.organizationId;
+
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = requesterTenant.data.features.find(
+        (feature) => feature.type === "PersistentCertifier"
+      )?.certifierId;
+
+      if (!certifierId) {
+        throw tenantIsNotACertifier(organizationId);
+      }
+
+      const attribute = await readModelService.getAttributeById(
+        unsafeBrandId(tenantAttributeSeed.id)
+      );
+
+      if (!attribute || attribute.kind !== "Certified") {
+        throw attributeNotFound(tenantAttributeSeed.id);
+      }
+
+      if (!attribute.origin || attribute.origin !== certifierId) {
+        throw certifiedAttributeOriginIsNotCompliantWithCertifier(
+          attribute.origin || "",
+          organizationId,
+          tenantId,
+          certifierId
+        );
+      }
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const certifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr) =>
+          attr.type === tenantAttributeType.CERTIFIED &&
+          attr.id === tenantAttributeSeed.id
+      ) as CertifiedTenantAttribute;
```

Comment:
I would prefer this approach here

```suggestion
      const certifiedTenantAttribute = targetTenant.data.attributes.find(
        (attr): attr is CertifiedTenantAttribute =>
          attr.type === tenantAttributeType.CERTIFIED &&
          attr.id === tenantAttributeSeed.id
      );
```